addons/website/models/website_visitor.py

```diff
@@ -185,7 +186,11 @@ def _get_visitor_from_request(self, force_create=False):
             # Cookie associated to a Partner
             visitor = Visitor
 
-        if force_create and not visitor:
+        if visitor and not visitor.timezone:
+            tz = self._get_visitor_timezone()
+            if tz:
+                visitor.timezone = tz
+        elif force_create and not visitor:
```

Comment:
Not a fan of elif, as actually the two statements are mainly independent. Would just say

if visitor and not visitor.timezone: blah
if not visitor and force_create: blah